src/lib/CLValue2/Option.ts

```diff
@@ -1,15 +1,35 @@
+import { Ok, Err, Some, None } from "ts-results";
 import { concat } from '@ethersproject/bytes';
 
 import { Option } from 'ts-results';
 
-import { CLValue, CLType, ToBytes } from './index';
+import { CLValue, CLType, 
+  ToBytes,
+  FromBytes,
+  CLErrorCodes,
+  ResultAndRemainder,
+  CLU8,
+  resultHelper
+} from './index';
 
 const OPTION_TAG_NONE = 0;
 const OPTION_TAG_SOME = 1;
 
-export class CLOptionType extends CLType {
+export class CLOptionType<T extends CLType | null> extends CLType {
```

Comment:
Event if the the Option is None it needs to have a type, so you have to remove that `null`. Constructor for none, could accept CLType to mark that.